netket/utils/static_range/range.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+
+class Range:
+    """
+    An object representing a range similar to python's range, but that
+    works with `jax.jit` and can be used within Numba-blocks.
+
+    This range object can also be used to convert 'computational basis'
+    configurations to integer indices ∈ [0,length].
+    """
+
+    def __init__(self, start, step, length, index_dtype=jnp.int64):
```

Comment:
I don't think that `index_dtype` should be a dtype of the range. 

In `basis_to_index` we can accept a `dtype` argument to specify the output dtype and `HilbertIndex` can use that one to coerce the output as they see fit.

Reply:
On the contrary, we could have a dtype attribute of the StaticRange if that is needed. 
If it's not needed, let's not add it.
Back in the day I had not added it because it's implicit in the dtype of start and step.